Sources/NIO/BaseSocketChannel.swift

```diff
@@ -646,8 +642,8 @@ class BaseSocketChannel<SocketType: BaseSocketProtocol>: SelectableChannel, Chan
             return
         }
 
-        executeAndComplete(promise) {
-            try socket.bind(to: address)
+        promise?.completeWithClosure {
```

Comment:
This is a subtle behavioural change and I'd be surprised if the tests don't pick it up: we _must_ execute this closure, but the code here will not execute if the promise is nil. I think the original variant is still necessary here.

Reply:
Yep, it's a subtle change. Let me take a look.